--- vtigerservice.php.orig	2006-08-02 15:58:50.000000000 -0600
+++ vtigerservice.php	2006-08-02 16:01:09.000000000 -0600
@@ -35,6 +35,14 @@
 	{
 		include("soap/thunderbirdplugin.php");
 	}
+	elseif($_REQUEST['service'] == "contact")
+	{
+		include("soap/contact.php");
+	}
+	elseif($_REQUEST['service'] == "fields")
+	{
+		include("soap/fields.php");
+	}
 	else
 	{
 		echo "No Service Configured for $_REQUEST[service]";
